pages/api/categories/index.js

```diff
@@ -0,0 +1,19 @@
+import dbConnect from "@/db/connect";
+import Category from "@/db/models/Category";
+
+export default async function handler(request, response) {
+  await dbConnect();
+
+  try {
+    if (request.method === "GET") {
+      const categories = await Category.find();
+      return response.status(200).json(categories);
+    } else if (request.method === "POST") {
+      const newCategory = JSON.parse(request.body);
+      const createdCategory = await Category.create(newCategory);
+      return response.status(201).json(createdCategory);
+    }
```

Comment:
since you return from the function when you hit the first if statement, you don't the the "else" and can add a nice empty line to make the code more readable:
```suggestion
    if (request.method === "GET") {
      const categories = await Category.find();
      response.status(200).json(categories);
      return 
    }
    // this empty line helps to visually separate the different request methods
    if (request.method === "POST") {
      const newCategory = request.body;
      const createdCategory = await Category.create(newCategory);
      response.status(201).json(createdCategory);
      return 
    }
```